cognite/client/_cognite_client.py

```diff
@@ -213,3 +217,64 @@ def default_oauth_interactive(
         """
         credentials = OAuthInteractive.default_for_azure_ad(tenant_id, client_id, cdf_cluster)
         return cls.default(project, cdf_cluster, credentials, client_name)
+
+    @classmethod
+    def from_dict(cls, config: dict[str, Any]) -> CogniteClient:
```

Comment:
I did some digging, and to keep this in line with the rest of the SDK, let's use load for both `dict` and `yaml`. I also suggest we don't support loading from a path-like object.

```py
@classmethod
    def load(cls, resource: dict | str) -> Self:
        ...
```

Taking a look at CogniteObject (in `_base.py`), consider adding: `dump` and `dump_yaml` methods 😄